core/txpool/txpool.go

```diff
@@ -1443,6 +1461,35 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	pool.shanghai = pool.chainconfig.IsShanghai(uint64(time.Now().Unix()))
 }
 
+func (pool *TxPool) validateMetaTxList(list *list) ([]*types.Transaction, *big.Int) {
+	currHeight := pool.chain.CurrentBlock().Number.Uint64()
+
+	var invalidMetaTxs []*types.Transaction
+	sponsorCostSum := big.NewInt(0)
+	for _, tx := range list.txs.Flatten() {
+		metaTxParams, err := types.DecodeAndVerifyMetaTxParams(tx)
+		if err != nil {
+			invalidMetaTxs = append(invalidMetaTxs, tx)
+			continue
+		}
+		if metaTxParams == nil {
+			continue
+		}
+		if metaTxParams.ExpireHeight < currHeight {
```

Comment:
What is the function of st.msg. MetaTxParams. ExpireHeight?

Reply:
The gas fee sponsor have no idea about when the users will submit meta transactions. Suppose the gas fee sponsor don't want to sponsor gas fee any more, and want to use the wallet in the future, it would be possible that sponsor wallet will be deducted native token in the future, which will be very confusing. With the `ExpireHeight`,  the sponsor can ensure that there will be no unconscious deduction after the highest `ExpireHeight`. 
In summary, the `ExpireHeight` can bring more determinism for the fee sponsor.